configuration/configmgr/configmgrlib/EnvironmentNode.cpp

```diff
@@ -217,66 +217,53 @@ std::string EnvironmentNode::getLocalValue() const
 }
 
 
-void EnvironmentNode::validate(Status &status, bool includeChildren, bool includeHiddenNodes) const
+void EnvironmentNode::validate(Status &status) const
 {
-    if (!m_pSchemaItem->isHidden() || includeHiddenNodes)
+    //
+    // Check node value
+    if (m_pLocalValue)
     {
-        //
-        // Check node value
-        if (m_pLocalValue)
-        {
-            m_pLocalValue->validate(status, m_id);
-        }
+        m_pLocalValue->validate(status, m_id);
+    }
+
+    //
+    // Check any attributes
+    for (auto &attrIt: m_attributes)
+    {
+        attrIt.second->validate(status, m_id);
 
         //
-        // Check any attributes
-        for (auto attrIt = m_attributes.begin(); attrIt != m_attributes.end(); ++attrIt)
+        // If this value must be unique, make sure it is
+        if (attrIt.second->getSchemaValue()->isUniqueValue())
         {
-            attrIt->second->validate(status, m_id);
-
-            //
-            // If this value must be unique, make sure it is
-            if (attrIt->second->getSchemaValue()->isUniqueValue())
+            bool found = false;
+            std::vector<std::string> allValues;
```

Comment:
Not required but allValues an uniqueValues could be hoisted outside of the for loop and cleared to avoid memory churn a bit.